### PR TITLE
React

### PR DESCRIPTION
**Pull request by PierreBarba**
# Hello reviewer I hope you are having a great day, any suggestion will be welcome, thank you very much for the support 😄 🌴 

## Ruby requirements
* Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
NOTE: Use Rails >=7 for this project so that you can follow the guidelines and instructions included in the curriculum. If you need help installing different versions of Rails, check this [tutorial](https://github.com/sinansevgi/reviewer-experiments/blob/main/specific_rails.md).

## Project requirements
* Create a new Rails app called 'hello-rails-react'.
    * Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
* Verify that the jsbundling-rails gem is added to your Gemfile.
* Initialize your project with Git.
* Install needed node dependencies as described in the previous lesson's tutorial.
* Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
* Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
* Make sure that your project has a Postgres database set up.
* Create a table for storing your messages and create 5 different greetings.
* Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
* Create a static view that will be the root of your app.

* Create the App component with react-router and render it in your static view.
* Create the Greeting component that will display a greeting. Set it up as a route in your App component.
* Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
* Display the random greeting in your Greeting component.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,222 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- SUPPORT -->
+
+## 👋 Show your support <a name="support"></a>
+
+Give a ⭐️ if you like this project!
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (BenMukebo) [inline README.md:174]: * [ ]  Dear @PierreBarba , Please remember we are not expected to have any broken links in our readme file. I would  recommend adding the issue link in the contributing sections: 

```
## 🤝 Contributing

Contributions, issues, and feature requests are welcome!

Feel free to check the [issues page](../../issues/).
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "app",
+  "private": "true",
+  "dependencies": {
+    "@hotwired/stimulus": "^3.2.1",
```
Comment (BenMukebo) [inline package.json:5]: - [ ] Kindly let's add these needed dependencies  `react-dom` and `babel-loader`  to your package.json file ✍🏻

File: app/models/message.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class Message < ApplicationRecord
```
Comment (BenMukebo) [inline app/models/message.rb:1]: - [ ] Here is the project requirement 👇🏻Kindly Make you create 5 different greetings, for the table for storing your messages, Actually you can use the seed to create five different greetings. ✍🏻 

image

File: babel.config.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (BenMukebo) [inline babel.config.js:1]: - [ ] Kindly, make sure your app runs correctly on your machine. 👍🏻 
- 🤔 I have followed the steps provided in the README and I'm still getting the same error. Please solve this 😄 

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @PierreBarba,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- There is a `Greeting` component. ✔️
- The `App`component has a route that leads to the `Greeting` component.
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._




Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (PierreBarba) [reply README.md:174]: @BenMukebo thanks for the observation i will fix this

File: package.json (hunk 2 above)
Comment (PierreBarba) [reply package.json:5]: @BenMukebo oh sorry but i have this dependency in my package.json 😕

File: app/models/message.rb (hunk 3 above)
Comment (PierreBarba) [reply app/models/message.rb:1]: @BenMukebo okay thanks i used just my api but i will create the seed thanks for the observation

File: babel.config.js (hunk 4 above)
Comment (PierreBarba) [reply babel.config.js:1]: @BenMukebo mmmm yeah in my computer its working very well but i will check thisCaptura de pantalla 2023-05-09 a la(s) 13 50 11

Captura de pantalla 2023-05-09 a la(s) 19 37 04

Captura de pantalla 2023-05-09 a la(s) 19 45 49

Captura de pantalla 2023-05-09 a la(s) 19 46 27


Captura de pantalla 2023-05-09 a la(s) 19 46 48

File: package.json (hunk 2 above)
Comment (BenMukebo) [reply package.json:5]: Yeah sure I can see that you do have `react-dom` but you're missing the `babel-loader`

File: babel.config.js (hunk 4 above)
Comment (BenMukebo) [reply babel.config.js:1]: - Great 👍🏻

File: README.md [hunk 5]
```diff
@@ -1 +1,196 @@
...
+<!-- CONTRIBUTING -->
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../../issues/).
+
```
Comment (thecodechaser) [inline README.md]: - [ ] As requested by the previous reviewer please make sure to fix the issues page link. It's still broken
 - The correct URL should be `https://github.com/PierreBarba/Ruby-on-Rails-React/issues`

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @PierreBarba👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @PierreBarba, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 

- PR is professional :+1: 
- Descriptive README.md file :+1: 
- App is working as expected :+1: 
- Linter checks passed :heavy_check_mark: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**